module-bluetooth/Bluetooth/error_bluetooth.cpp

```diff
@@ -0,0 +1,138 @@
+        return "ERROR_CODE_LMP_PDU_NOT_ALLOWED";
+    case ERROR_CODE_ENCRYPTION_MODE_NOT_ACCEPTABLE:
+        return "ERROR_CODE_ENCRYPTION_MODE_NOT_ACCEPTABLE";
+    case ERROR_CODE_LINK_KEY_CANNOT_BE_CHANGED:
+        return "ERROR_CODE_LINK_KEY_CANNOT_BE_CHANGED";
+    case ERROR_CODE_REQUESTED_QOS_NOT_SUPPORTED:
+        return "ERROR_CODE_REQUESTED_QOS_NOT_SUPPORTED";
+    case ERROR_CODE_INSTANT_PASSED:
+        return "ERROR_CODE_INSTANT_PASSED";
+    case ERROR_CODE_PAIRING_WITH_UNIT_KEY_NOT_SUPPORTED:
+        return "ERROR_CODE_PAIRING_WITH_UNIT_KEY_NOT_SUPPORTED";
+    case ERROR_CODE_DIFFERENT_TRANSACTION_COLLISION:
+        return "ERROR_CODE_DIFFERENT_TRANSACTION_COLLISION";
+    case ERROR_CODE_RESERVED:
+        return "ERROR_CODE_RESERVED";
+    case ERROR_CODE_QOS_UNACCEPTABLE_PARAMETER:
+        return "ERROR_CODE_QOS_UNACCEPTABLE_PARAMETER";
+    case ERROR_CODE_QOS_REJECTED:
+        return "ERROR_CODE_QOS_REJECTED";
+    case ERROR_CODE_CHANNEL_CLASSIFICATION_NOT_SUPPORTED:
+        return "ERROR_CODE_CHANNEL_CLASSIFICATION_NOT_SUPPORTED";
+    case ERROR_CODE_INSUFFICIENT_SECURITY:
+        return "ERROR_CODE_INSUFFICIENT_SECURITY";
+    case ERROR_CODE_PARAMETER_OUT_OF_MANDATORY_RANGE:
+        return "ERROR_CODE_PARAMETER_OUT_OF_MANDATORY_RANGE";
+    case ERROR_CODE_ROLE_SWITCH_PENDING:
+        return "ERROR_CODE_ROLE_SWITCH_PENDING";
+    case ERROR_CODE_RESERVED_SLOT_VIOLATION:
+        return "ERROR_CODE_RESERVED_SLOT_VIOLATION";
+    case ERROR_CODE_ROLE_SWITCH_FAILED:
+        return "ERROR_CODE_ROLE_SWITCH_FAILED";
+    case ERROR_CODE_EXTENDED_INQUIRY_RESPONSE_TOO_LARGE:
+        return "ERROR_CODE_EXTENDED_INQUIRY_RESPONSE_TOO_LARGE";
+    case ERROR_CODE_SECURE_SIMPLE_PAIRING_NOT_SUPPORTED_BY_HOST:
+        return "ERROR_CODE_SECURE_SIMPLE_PAIRING_NOT_SUPPORTED_BY_HOST";
+    case ERROR_CODE_HOST_BUSY_PAIRING:
+        return "ERROR_CODE_HOST_BUSY_PAIRING";
+    case ERROR_CODE_CONNECTION_REJECTED_DUE_TO_NO_SUITABLE_CHANNEL_FOUND:
+        return "ERROR_CODE_CONNECTION_REJECTED_DUE_TO_NO_SUITABLE_CHANNEL_FOUND";
+    case ERROR_CODE_CONTROLLER_BUSY:
+        return "ERROR_CODE_CONTROLLER_BUSY";
+    case ERROR_CODE_UNACCEPTABLE_CONNECTION_PARAMETERS:
+        return "ERROR_CODE_UNACCEPTABLE_CONNECTION_PARAMETERS";
+    case ERROR_CODE_DIRECTED_ADVERTISING_TIMEOUT:
+        return "ERROR_CODE_DIRECTED_ADVERTISING_TIMEOUT";
+    case ERROR_CODE_CONNECTION_TERMINATED_DUE_TO_MIC_FAILURE:
+        return "ERROR_CODE_CONNECTION_TERMINATED_DUE_TO_MIC_FAILURE";
+    case ERROR_CODE_CONNECTION_FAILED_TO_BE_ESTABLISHED:
+        return "ERROR_CODE_CONNECTION_FAILED_TO_BE_ESTABLISHED";
+    case ERROR_CODE_MAC_CONNECTION_FAILED:
+        return "ERROR_CODE_MAC_CONNECTION_FAILED";
+    case ERROR_CODE_COARSE_CLOCK_ADJUSTMENT_REJECTED_BUT_WILL_TRY_TO_ADJUST_USING_CLOCK_DRAGGING:
+        return "ERROR_CODE_COARSE_CLOCK_ADJUSTMENT_REJECTED_BUT_WILL_TRY_TO_ADJUST_USING_CLOCK_DRAGGING";
+    default:
```

Comment:
do you need this default? consider simply covering all cases

Reply:
or maybe use magic enum

Reply:
@alekrudnik this is just a subset of codes + library has its own defined ones. This should suffice for now for our needs in terms - something is better than nothing.
We cant use magic enum - as these are not enums. These are defines :)

Reply:
like @pholat said, those codes are defines in stack header - we are not able to convert it to enums/strings automatically.